src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -302,7 +305,20 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                 logging.debug("Updated job summary state history for job %s" % jobSummaryId)
 
             if job.get("fwjr", None):
-
+                cached_workflow = self.workflow_cache.setdefault(job['workflow'], {})
+                if job['task'] not in cached_workflow:
+                    workload = WMWorkloadHelper()
+                    workload.load(self.getWorkflowSpecDAO.execute(job['task'])['spec'])
```

Comment:
@ticoann @amaltaro
This line is failing in some tests with the error that `'spec'` is a `KeyError`. I don't understand how this can be as the relevant DAO call should include `'spec'`, see link below:
[GetSpecAndNameFromTask.py#L47](https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMBS/MySQL/Workflow/GetSpecAndNameFromTask.py#L47)

Any ideas?